src/provider/provider_level_zero.c

```diff
@@ -347,12 +354,29 @@ static umf_result_t ze_memory_provider_open_ipc_handle(void *provider,
     ASSERT(providerIpcData != NULL);
     ASSERT(ptr != NULL);
     ze_result_t ze_result;
-    ze_ipc_mem_handle_t *ze_ipc_handle = (ze_ipc_mem_handle_t *)providerIpcData;
+    ze_ipc_data_t *ze_ipc_data = (ze_ipc_data_t *)providerIpcData;
     struct ze_memory_provider_t *ze_provider =
         (struct ze_memory_provider_t *)provider;
+    int fd_local = -1;
+    ze_ipc_mem_handle_t ze_ipc_handle = ze_ipc_data->ze_handle;
+
+    if (ze_ipc_data->pid != utils_getpid()) {
+        int fd_remote = -1;
+        memcpy(&fd_remote, &ze_ipc_handle, sizeof(fd_remote));
+        umf_result_t umf_result =
+            utils_duplicate_fd(ze_ipc_data->pid, fd_remote, &fd_local);
+        if (umf_result != UMF_RESULT_SUCCESS) {
+            LOG_PERR("duplicating file descriptor failed");
+            return umf_result;
+        }
+        memcpy(&ze_ipc_handle, &fd_local, sizeof(fd_local));
+    }
 
     ze_result = g_ze_ops.zeMemOpenIpcHandle(
-        ze_provider->context, ze_provider->device, *ze_ipc_handle, 0, ptr);
+        ze_provider->context, ze_provider->device, ze_ipc_handle, 0, ptr);
+    if (fd_local != -1) {
+        utils_close_fd(fd_local);
```

Comment:
`(void)utils_close_fd(args)`?
It returns an integer value, and in provider_os_memory.c:329, such casting is left as is.